sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveHintsSuite.scala

```diff
@@ -294,4 +294,50 @@ class ResolveHintsSuite extends AnalysisTest {
         caseSensitive = true)
     }
   }
+
+  test("disable hint resolution when spark.sql.optimizer.hints.enabled = false") {
+    checkAnalysis(
+      UnresolvedHint("COALESCE", Seq(Literal(10)), table("TaBlE")),
+      testRelation,
+      caseSensitive = true,
+      enableHints = false
```

Comment:
Could you use `withSQLConf` instead for the tests? I think `enableHints` is only used for this test suite, so we don't need to add the option in `AnalysisTest`.

Reply:
@maropu Should i move this test to SQLQuerySuite ? Actually i tried initially to do withSQLConf, but realized that this suite creates an Analyzer by constructing a SQLConf inside AnalysisTest.

Reply:
Ah, yea. that sounds better.